pkg/apis/core/validation/validation_test.go

```diff
@@ -13498,3 +13575,541 @@ func TestAlphaVolumePVCDataSource(t *testing.T) {
+				ObjectMeta: metav1.ObjectMeta{Name: "single-ip-family-6", Namespace: "ns"},
+				Spec: core.PodSpec{
+					Containers: []core.Container{
+						{
+							Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: "File",
+						},
+					},
+					RestartPolicy: core.RestartPolicyAlways,
+					DNSPolicy:     core.DNSClusterFirst,
+				},
+				Status: core.PodStatus{
+					PodIPs: []core.PodIP{
+						{IP: "::1"},
+					},
+				},
+			},
+		},
+		{
+			expectError:     false,
+			enableDualStack: false,
+			pod: core.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "single-ip-family-6", Namespace: "ns"},
+				Spec: core.PodSpec{
+					Containers: []core.Container{
+						{
+							Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: "File",
+						},
+					},
+					RestartPolicy: core.RestartPolicyAlways,
+					DNSPolicy:     core.DNSClusterFirst,
+				},
+				Status: core.PodStatus{
+					PodIPs: []core.PodIP{
+						{IP: "1.1.1.1"},
+					},
+				},
+			},
+		},
+		{
+			expectError:     false,
+			enableDualStack: true,
+			pod: core.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "dual-stack-4-6", Namespace: "ns"},
+				Spec: core.PodSpec{
+					Containers: []core.Container{
+						{
+							Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: "File",
+						},
+					},
+					RestartPolicy: core.RestartPolicyAlways,
+					DNSPolicy:     core.DNSClusterFirst,
+				},
+				Status: core.PodStatus{
+					PodIPs: []core.PodIP{
+						{IP: "1.1.1.1"},
+						{IP: "::1"},
+					},
+				},
+			},
+		},
+		{
+			expectError:     false,
+			enableDualStack: true,
+			pod: core.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "dual-stack-6-4", Namespace: "ns"},
+				Spec: core.PodSpec{
+					Containers: []core.Container{
+						{
+							Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: "File",
+						},
+					},
+					RestartPolicy: core.RestartPolicyAlways,
+					DNSPolicy:     core.DNSClusterFirst,
+				},
+				Status: core.PodStatus{
+					PodIPs: []core.PodIP{
+						{IP: "::1"},
+						{IP: "1.1.1.1"},
+					},
+				},
+			},
+		},
+		/* failure cases start here */
+		{
+			expectError:     true,
+			enableDualStack: false,
+			pod: core.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "invalid-pod-ip", Namespace: "ns"},
+				Spec: core.PodSpec{
+					Containers: []core.Container{
+						{
+							Name: "ctr", Image: "image", ImagePullPolicy: "IfNotPresent", TerminationMessagePolicy: "File",
+						},
+					},
+					RestartPolicy: core.RestartPolicyAlways,
+					DNSPolicy:     core.DNSClusterFirst,
+				},
+				Status: core.PodStatus{
+					PodIPs: []core.PodIP{
+						{IP: "this-is-not-an-ip"},
+					},
+				},
+			},
+		},
+		{
+			expectError:     true,
+			enableDualStack: false,
+			pod: core.Pod{
+				ObjectMeta: metav1.ObjectMeta{Name: "dualstack-with-dualstack-feature-not-enabled", Namespace: "ns"},
```

Comment:
not needed if we don't gate check

Reply:
The test runs with the gate turned off. and expectedly fail because of the same the reason. i forced the gate value at the beginning of every test, to make it clearer.